vizro-core/src/vizro/models/_components/form/_form_utils.py

```diff
@@ -100,3 +101,37 @@ def set_default_marks(cls, marks, values):
     if not marks and values.get("step") is None:
         marks = None
     return marks
+
+
+def parse_date(date_str):
+    """Parse date string to datetime object."""
+    return datetime.strptime(date_str, "%Y-%m-%d")
+
+
+def validate_date_picker_value(cls, value, values):
+    """Reusable validator for the "value" argument for sliders."""
+    if value is None:
+        return value
+
+    start_date, end_date = (value[0], value[1]) if isinstance(value, list) else (value, value)
+    start_date, end_date = parse_date(start_date), parse_date(end_date)
+
+    if start_date >= end_date:
+        raise ValueError("Please provide a valid date range for value.")
+
+    if (values["min_date"] is not None and not start_date >= parse_date(values["min_date"])) or (
+        values["max_date"] is not None and not end_date <= parse_date(values["max_date"])
+    ):
+        raise ValueError("Please provide a valid value between the min and max value.")
+
+    return value
```

Comment:
Similar comment as below - can we just refactor the `validate_slider_value` function and then re-use here?